client/packages/electron/src/electron.ts

```diff
@@ -7,12 +7,59 @@ import {
   frontEndHostUrl,
   isProtocol,
 } from '@openmsupply-client/common/src/hooks/useNativeClient';
+import HID from 'node-hid';
+
 const SERVICE_TYPE = 'omsupply';
 const PROTOCOL_KEY = 'protocol';
 const CLIENT_VERSION_KEY = 'client_version';
 const HARDWARE_ID_KEY = 'hardware_id';
+const SUPPORTED_SCANNERS = [
+  {
+    vendorId: 1504,
+    vendorName: 'Zebra',
+    products: [{ id: 2194, model: 'DS2208' }],
+  },
+];
+
+class BarcodeScanner {
+  device: HID.HID | undefined;
+
+  constructor() {
+    this.findDevice();
+  }
+
+  findDevice() {
+    const devices = HID.devices();
+    for (const scanner of SUPPORTED_SCANNERS) {
+      const productIds = scanner.products.map(p => p.id);
+      const deviceInfo = devices.find(d => {
+        d.vendorId === scanner.vendorId &&
+          productIds.some(pid => d.productId === pid);
+      });
+      if (deviceInfo && !!deviceInfo.path) {
+        this.device = new HID.HID(deviceInfo.path);
+        break;
+      }
+    }
+  }
+
+  start() {
+    return new Promise((resolve, reject) => {
+      this.device?.read((err, data) => {
+        if (err) reject(err);
+        resolve(data);
+      });
```

Comment:
This promise never resolves if this.device is undefined, i.e. call reject if there is an error